### PR TITLE
#DP-62 (Feat): Payment Forms

### PR DESCRIPTION
**Pull request by Dev-nkundineza**
### PR Description
This PR is for the payment forms, it adds the multi step form that pro users can use to add payments. 
### Description of tasks that were expected to be completed

- [x] Well designed multi-step Form 
- [x] Responsive on all devices  
- [x] The form fields are validated

### How has this been tested?
```
git clone https://github.com/atlp-rwanda/atlp-pulse-fn.git 
git checkout ft-multistepForm-dp-62
run npm install
checkout branch ft-multistepForm-dp-62
 npm run dev
```
### Screenshots (If appropriate)
[screencapture-localhost-3000-pricing-form-2022-08-03-16_59_35](https://user-images.githubusercontent.com/79087014/182641350-d622ce14-8510-43ef-a752-56ebd68305e4.png)

### CODE REVIEW

**Comment by feyton** [comment]
I deployed a review app for you to try out this pull request 👇🏽

Heroku App: https://devpulse-review-app-53.herokuapp.com/


<hidden purpose="for-rewritable-pr-comment-action-use" value="Deploy"></hidden>

**Review by samyshyaka — CHANGES_REQUESTED** [review]
You did a great job @Dev-nkundineza,
- The user shouldn't be able to go the next section without filling the form
- On the payment section, try to make sure that the expiration input is vertically aligned with the card holder name input. This also goes to the CVV input with the Card number input.

**Comment by Dev-nkundineza** [comment]
> You did a great job @Dev-nkundineza,
> 
> * The user shouldn't be able to go the next section without filling the form
> * On the payment section, try to make sure that the expiration input is vertically aligned with the card holder name input. This also goes to the CVV input with the Card number input.

hello @samyshyaka , re-review the changes

**Review by chanceDira — DISMISSED** [review]
Nice work @Dev-nkundineza

**Review by PatrickTUy — DISMISSED** [review]
Good work, let's have the few percentage that dropped increased

File: src/Routes.tsx [hunk 1]
```diff
@@ -8,6 +8,7 @@ import Pricing from './pages/Pricing';
 import PricingForm from './components/PricingForm';
 import AdminLogin from './pages/Organization/AdminLogin';
 import Perfomancetraineetable from './components/TraineePerformance';
+import Noredirect from './pages/Noredirect';
 
 function MainRoutes() {
   return (
```
Comment (feyton) [inline src/Routes.tsx:14]: We are missing the route to go to the `pricing` page.

Please fix the issue so that we can test the feature

**Review by feyton — CHANGES_REQUESTED** [review]
Fix the issue of missing route that redirect to the **`pricing`** page

**Review by feyton — APPROVED** [review]

**Review by ceelogre — CHANGES_REQUESTED** [review]
How are we to test this?

**Comment by Dev-nkundineza** [comment]
> How are we to test this?

you can use the deployed review app https://devpulse-review-app-53.herokuapp.com/ 

and navigate to:
https://devpulse-review-app-53.herokuapp.com/pricing-form

**Review by ceelogre — CHANGES_REQUESTED** [review]
There's no need to show an error message before a user fills out a form.
Screenshot_2022-08-16_17-29-13


This form is not properly aligned.
Screenshot_2022-08-16_17-30-10

There's a mix up of languages
IMG_9962

**Comment by Dev-nkundineza** [comment]
> https://devpulse-review-app-53.herokuapp.com/pricing-form

issues Resolved, kindly re-review, thanks.

**Comment by Dev-nkundineza** [comment]
> There's no need to show an error message before a user fills out a form. Screenshot_2022-08-16_17-29-13
> 
> This form is not properly aligned. Screenshot_2022-08-16_17-30-10
> 
> There's a mix up of languages IMG_9962

resolved

**Review by rukundo-kevin — CHANGES_REQUESTED** [review]
Great work @Dev-nkundineza ,
Just few untranslated contents when in Kinyarwanda.
Screen Shot 2022-08-17 at 13 06 16
Screen Shot 2022-08-17 at 13 08 31

**Review by rukundo-kevin — APPROVED** [review]

**Comment by Dev-nkundineza** [comment]
> Great work @Dev-nkundineza , Just few untranslated contents when in Kinyarwanda. Screen Shot 2022-08-17 at 13 06 16 Screen Shot 2022-08-17 at 13 08 31

issues solved, thanks

**Review by ceelogre — CHANGES_REQUESTED** [review]
This first step still has issues. You cannot let users input the countries, it would have to be a drop-down which implies that you cannot have them input provinces and this adds an unnecessary task of adding all countries and regions/provinces. I suggest you just get the org name, street address, city and optionally postal code.  
payment form

On the second step, expiration date is usually one field, so I suggest you merge them and then center the form. This applies to the first step too.
pay 2

**Review by ceelogre — CHANGES_REQUESTED** [review]
Work with Jean on how to validate a card.

**Review by ceelogre — CHANGES_REQUESTED** [review]
CCV should be 3 digits.
cvv

**Comment by Dev-nkundineza** [comment]
> CCV should be 3 digits. cvv

Resolved.

File: src/components/Payment-steps/steps/Account.tsx [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+  const [onClickCard, setOnClickCard] = useState({
+    creditCard: false,
+    paypal: false,
+    applepay: false,
+    onClickState: false,
+  });
+  const { userData, setUserData, formValidation, displayError, valid } =
+    useStepperContext();
+
+  console.log(userData);
```
Comment (feyton) [inline src/components/Payment-steps/steps/Account.tsx:19]: Can we remove the unnecessary consoles here when push to remote

**Review by feyton — CHANGES_REQUESTED** [review]
@Dev-nkundineza 
Can we also display the errors on user inputs

File: src/components/Payment-steps/steps/Account.tsx (hunk 2 above)
Comment (ceelogre) [reply src/components/Payment-steps/steps/Account.tsx:19]: I thought we have hooks to block `console` messages when pushing?

Comment (rukundo-kevin) [reply src/components/Payment-steps/steps/Account.tsx:19]: > I thought we have hooks to block `console` messages when pushing?

ESLint would have blocked it but there's seems to be `eslint-disable`

**Review by feyton — APPROVED** [review]

**Review by jeandamascyifuzo — APPROVED** [review]
well done

**Review by rukundo-kevin — CHANGES_REQUESTED** [review]
@Dev-nkundineza , @feyton Great job,
Just translation on validation messages left.
Screen Shot 2022-08-28 at 09 02 38
Screen Shot 2022-08-28 at 09 03 23

**Comment by Dev-nkundineza** [comment]
> @Dev-nkundineza , @feyton Great job, Just translation on validation messages left. Screen Shot 2022-08-28 at 09 02 38 Screen Shot 2022-08-28 at 09 03 23

resolved..

**Review by rukundo-kevin — CHANGES_REQUESTED** [review]
Still some minor translations.
Screen Shot 2022-08-28 at 10 40 33

Screen Shot 2022-08-28 at 10 39 11

**Review by PatrickTUy — APPROVED** [review]
Approved

**Comment by Dev-nkundineza** [comment]
> Still some minor translations. 
resolved

**Review by rukundo-kevin — APPROVED** [review]
Great work @Dev-nkundineza

**Review by ceelogre — APPROVED** [review]

**Pull request merged**